### PR TITLE
Review request for microsoft.support 2024-04-01

### PR DESCRIPTION
**Pull request by raghumunukutla**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
Please complete PR description, purpose of PR, due diligence sections

**Comment by TimLovellSmith** [comment]
>           "description": "Enrollment Id associated with the support ticket.",

optional, is it?

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:2466 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Size of each chunk",

in ... bytes??

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:3027 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Size of the file to be uploaded",

in ... bytes??

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:3032 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>     }

Why not use the parameter definitions from common-types?

You currently seem to be getting linter errors like these, as a side effect or not doing so. Could you just fix this to use common types?

> The parameter 'ApiVersionParameter' is defined in global parameters section without 'x-ms-parameter-location' extension. This would add the parameter as the client property. Please ensure that this is exactly you want. If so, apply the extension 'x-ms-parameter-location': 'client'. Else, apply the extension 'x-ms-parameter-location': 'method'.
Location: Microsoft.Support/stable/2024-04-01/support.json#L3067

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:3073 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "Checks whether name is available for SupportTicket resource": {

Linter says you have duplicate descriptions
'Checks whether name is available for SupportTicket resource'

This is a sign you need to improve your descriptions

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:247 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "Checks whether name is available for SupportTicket resource": {

Linter says you have duplicate descriptions
'Checks whether name is available for SupportTicket resource'

This is a sign you need to improve your descriptions

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:292 in 33f65df. [](commit_id = 33f65df5bcd15bd8efadeaf17c728e6c06149819, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "Checks whether name is available for SupportTicket resource": {

You also have a bunch more of these, see LintDiff output.

---
In reply to: [1986225812](https://github.com/Azure/azure-rest-api-specs/pull/28068#issuecomment-1986225812) [](http://example.com/codeflow?ancestors=1986225812)

---
Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:292 in e344df5. [](commit_id = e344df5ec324be8cb924f8c14489b0c3c6eefc43, deletion_comment = False)

File: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/examples/ListSupportTicketsBySubscription.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            "id": "/subscriptions/subid/providers/Microsoft.Support/supportTickets/testticket2",
+            "name": "testticket2",
+            "type": "Microsoft.Support/supportTickets",
+            "properties": {
+              "supportTicketId": "118032014183771",
+              "description": "This is a test - please ignore",
+              "problemClassificationId": "/providers/Microsoft.Support/services/subscription_management_service_guid/problemClassifications/problemClassification_guid",
+              "problemClassificationDisplayName": "Add or Edit VAT, TAX ID, or PO Number",
+              "severity": "minimal",
+              "require24X7Response": false,
```
Comment (TimLovellSmith) [inline specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/examples/ListSupportTicketsBySubscription.json:71]: >require24X7Response [](http://example.com/codeflow?start=15&length=19)

Missing required property: advancedDiagnosticConsent
Url: Microsoft.Support/stable/2024-04-01/support.json#L2421:39
ExampleUrl: stable/2024-04-01/examples/ListSupportTicketsBySubscription.json

Comment (TimLovellSmith) [reply specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/examples/ListSupportTicketsBySubscription.json:71]: (I noticed this by reviewing ModelValidation errors.) Please fix modelVAlidation errors!

**Comment by raghumunukutla** [comment]
> > ```
> >       "description": "Enrollment Id associated with the support ticket.",
> > ```
> 
> optional, is it?
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:2466 in [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819). [](commit_id = [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819), deletion_comment = False)

Yes, this is an optional property.

**Comment by raghumunukutla** [comment]
> Size of each chunk

Current supported max chunk size is 2.5 mb. Max limit of int32 is 2147 MB which is way larger than the need, so kept it for int32

**Comment by raghumunukutla** [comment]
> Please complete PR description, purpose of PR, due diligence sections

Updated

**Comment by raghumunukutla** [comment]
> > ```
> >       "description": "Size of each chunk",
> > ```
> 
> in ... bytes??
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:3027 in [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819). [](commit_id = [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819), deletion_comment = False)

Updated the description

**Comment by raghumunukutla** [comment]
> bytes?

Updated the swagger to use common types.

File: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/examples/ListSupportTicketsBySubscription.json (hunk 1 above)
Comment (raghumunukutla) [reply specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/examples/ListSupportTicketsBySubscription.json:71]: Taken care of.

**Comment by raghumunukutla** [comment]
> > ```
> >       "Checks whether name is available for SupportTicket resource": {
> > ```
> 
> You also have a bunch more of these, see LintDiff output.
> 
> In reply to: [1986225812](https://github.com/Azure/azure-rest-api-specs/pull/28068#issuecomment-1986225812)
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:292 in [e344df5](https://github.com/Azure/azure-rest-api-specs/commit/e344df5ec324be8cb924f8c14489b0c3c6eefc43). [](commit_id = [e344df5](https://github.com/Azure/azure-rest-api-specs/commit/e344df5ec324be8cb924f8c14489b0c3c6eefc43), deletion_comment = False)

Taken care of.

**Comment by raghumunukutla** [comment]
> > ```
> >       "Checks whether name is available for SupportTicket resource": {
> > ```
> 
> Linter says you have duplicate descriptions 'Checks whether name is available for SupportTicket resource'
> 
> This is a sign you need to improve your descriptions
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:247 in [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819). [](commit_id = [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819), deletion_comment = False)

Taken care of

**Comment by raghumunukutla** [comment]
> > ```
> >       "Checks whether name is available for SupportTicket resource": {
> > ```
> 
> Linter says you have duplicate descriptions 'Checks whether name is available for SupportTicket resource'
> 
> This is a sign you need to improve your descriptions
> 
> Refers to: specification/support/resource-manager/Microsoft.Support/stable/2024-04-01/support.json:247 in [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819). [](commit_id = [33f65df](https://github.com/Azure/azure-rest-api-specs/commit/33f65df5bcd15bd8efadeaf17c728e6c06149819), deletion_comment = False)

Taken care of

**Review by visingla-ms — APPROVED** [review]

**Pull request merged**